src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,31 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        User existingUser = storageDao.get(user.getLogin());
+        if (existingUser != null) {
+            throw new InvalidUserDataException("User with this login already exists");
+        }
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (user.getLogin() == null || user.getLogin().length() < MIN_LOGIN_LENGTH
```

Comment:
It is not clear what exactly is wrong with the user in case of an error. It is better to separately check the age, password length, and so on. And display the corresponding error text.